reactivesocket-core/src/main/java/io/reactivesocket/util/PayloadImpl.java

```diff
@@ -61,17 +71,39 @@ public PayloadImpl(ByteBuffer data) {
     }
 
     public PayloadImpl(ByteBuffer data, ByteBuffer metadata) {
+        this(data, metadata, false);
+    }
+
+    /**
+     * New instance where every invocation of {@link #getMetadata()} and {@link #getData()} will reset the position of
+     * the buffer to the position when it is created, if and only if, {@code singleUse} is set to {@code false}.
+     *
+     * @param data Buffer for data.
+     * @param metadata Buffer for metadata.
+     * @param singleUse {@code true} if the buffer position is to be reset on every invocation of {@link #getData()} and
+     * {@link #getMetadata()}.
+     */
+    public PayloadImpl(ByteBuffer data, ByteBuffer metadata, boolean singleUse) {
```

Comment:
nitpick: but renaming/reversing `singleUse` to `reusable` may be easier to understand (especially if you don't read the doc).